src/server/api.js

```diff
@@ -0,0 +1,372 @@
+      {
+        valueFormat: NUMBER_FORMAT_SHORT_SUFFIXED,
+        optionalValue: averageVolume,
+        optionalValueFormat: NUMBER_FORMAT_SHORT_SUFFIXED
+      }
+    ),
+    [LABEL_PE_RATIO]: createStockDataEntry(
+      trailingPE,
+      {
+        optionalValue: trailingEps
+      }
+    ),
+    [LABEL_ROE]: createStockDataEntry(
+      returnOnEquity,
+      {
+        valueFormat: NUMBER_FORMAT_PERCENT
+      }
+    ),
+    [LABEL_FCFY]: createStockDataEntry(
+      calculateFcfy(freeCashflow, marketCap),
+      {
+        valueFormat: NUMBER_FORMAT_PERCENT
+      }
+    )
+  };
+};
+
+const createStock = stockQuote => {
+  const {
+    price,
+    summaryDetail,
+    financialData,
+    defaultKeyStatistics
+  } = stockQuote;
+  return {
+    companyName: price.shortName,
+    symbol: price.symbol,
+    exchange: price.exchangeName,
+    stockOverviewData: processStockData(
+      Object.assign(
+        {},
+        summaryDetail,
+        financialData,
+        defaultKeyStatistics
+      )
+    )
+  };
+};
+
+// Used for historical() data obtained using period 'd'
+const getDatesAndPrices = dailyData => {
+  let datesAndPrices = [];
+  dailyData.forEach(({
+    date,
+    close
+  }) => {
+    datesAndPrices.unshift({
+      date: formatDateForMaxStockData(date),
+      price: close
+    });
+  });
+  return datesAndPrices;
+};
+
+// NOTE: this date has timezone UTC, which is incorrect but works in this
+// case because we are just extracting the time
+const getAdjustedDateForTimestamp = (gmtoffset, timestamp) => {
+  const adjustedTimestamp =
+    (timestamp + gmtoffset) * MILLISECONDS_PER_SECOND;
+  return new Date(adjustedTimestamp);
+};
+
+const getDateAndTime = (gmtoffset, timestamp, dateAndTimeFormat) => {
+  const dateAndTime = getAdjustedDateForTimestamp(gmtoffset, timestamp);
+  return dateFormat(dateAndTime, dateAndTimeFormat, true);
+};
+
+const getStartOfDayTimestampIndex = (dayIndex, timestampsPerDay) =>
+  Math.floor(dayIndex * timestampsPerDay);
+
+const getEndOfDayTimestampIndex = (dayIndex, timestampsPerDay) =>
+  Math.floor((dayIndex + 1) * timestampsPerDay);
+
+const getDatesAndTimesForOneDay = (
+  close,
+  dayIndex,
+  gmtoffset,
+  numberOfDays,
+  timestamp,
+  timestampIntervals
+) => {
+  let datesTimesAndPrices = [];
+  const timestampsPerDay = timestamp.length / numberOfDays;
+  const dateAndTimeFormat = (numberOfDays === ONE_DAY)
+    ? DATE_FORMAT_ONE_DAY
+    : DATE_FORMAT_FIVE_DAY;
+
+  for (
+    let i = getStartOfDayTimestampIndex(dayIndex, timestampsPerDay);
+    i < getEndOfDayTimestampIndex(dayIndex, timestampsPerDay);
+    i++
+  ) {
+    if (timestamp[i] < timestampIntervals[dayIndex].start
+      || timestamp[i] > timestampIntervals[dayIndex].end) {
+      continue;
+    }
+    datesTimesAndPrices.push({
+      dateAndTime: getDateAndTime(gmtoffset, timestamp[i], dateAndTimeFormat),
+      price: close[i]
+    });
+  }
+  return datesTimesAndPrices;
+};
+
+// days are 0 indexed
```

Comment:
What do you think about 
```javascript
const getTimestampForDay = (dayIndex, meta, isStart=true) =>
  // regular consists of an array of arrays, where the first array
  // index corresponds to the day, second is always a single element array
  meta.tradingPeriods.regular[dayIndex][0][isStart ? 'start' : 'end'];
}
```
or 
```javascript
const getTimestampForDay = (dayIndex, meta, time='start') =>
  // regular consists of an array of arrays, where the first array
  // index corresponds to the day, second is always a single element array
  meta.tradingPeriods.regular[dayIndex][0][time];
}
```
Not sure if one of these is better, but worth thinking about because `getEndTimestampForDay` and `getStartTimestampForDay` are basically the same.